code/workspaces/web-app/src/hooks/assetRepoHooks.js

```diff
@@ -1,5 +1,19 @@
 import useShallowSelector from './useShallowSelector';
 import assetRepoSelectors from '../selectors/assetRepoSelectors';
+import sortByName from '../components/common/sortByName';
 
-// eslint-disable-next-line import/prefer-default-export
 export const useAssetRepo = () => useShallowSelector(assetRepoSelectors.assetRepo);
+
+export const useVisibleAssets = (projectKey) => {
+  const assetRepo = useAssetRepo();
+  const visibleAssets = assetRepo.value.assets
+    ? assetRepo.value.assets.filter(asset => asset.visible === 'PUBLIC' || asset.projects.includes(projectKey))
+    : [];
```

Comment:
This logic is handled in the infrastructure api. There are two available calls, one to list all of the assets (for which the user has to be a data manger) and one for the assets that a project has access to which does this filtering.

Reply:
That's true, and that ensures that a non-administrator never gets assets from the client API that aren't visible for their project.
However, there's another scenario - an admin may be looking at Admin > Resources, where there are also Notebook Edit buttons for multiple projects.  In this case, the redux state will have assets from multiple projects - but it's still the case that, for a particular notebook, the edit dialog should only list assets visible for that project.  Hence the need to also have the logic here.

Reply:
Ahhh I see! Good thinking Andy! Thanks